src/charm.py

```diff
@@ -117,6 +123,55 @@ def _on_config_changed(self, event: ops.HookEvent) -> None:
         """
         self.change_config(event)
         self._set_workload_version()
+        if self._charm_state.enable_mjolnir:
+            self._enable_mjolnir()
+
+    def _enable_mjolnir(self) -> None:
+        """Enable mjolnir service."""
+        container = self.unit.get_container(SYNAPSE_CONTAINER_NAME)
+        if not container.can_connect():
+            self.unit.status = ops.MaintenanceStatus("Waiting for pebble")
+            return
+        self.model.unit.status = ops.MaintenanceStatus("Configuring Mjolnir")
+        try:
+            admin_username = token_hex(16)
+            admin_user = actions.register_user(container, admin_username, True)
```

Comment:
I'm wondering if this user shouldn't be systematically created when the charm is start and kept for the whole lifecycle of the application.
It's very likely that we'll need that user for many actions, and creation/disabling a user systematically is probably not the best approach.

Reply:
I've just realized while testing that if we deactivate the account used to get the access token, the access token is deleted as well so this internal admin should be kept. How can we store its password?

Reply:
I'd rather go for option 2
Either store it in peer relation data, or on the shared storage between the charm and the workload (don't remember where this is mounted)

Option 1 will keep creating admin users, this is not great

Reply:
Agree, changed.